### PR TITLE
WIP: Fix webdav for davfs2

### PR DESCRIPTION
**Pull request by inkhey**
davfs2 need a working "LOCK" (Doing a lock create automatically an empty file) . As method to create empty file doesn't return a 'File' type, automatic creation of new file when "LOCK" inexistent file in wsgidav failed. 

Return a valid File in method to create empty file resolve this issue.

### CODE REVIEW

File: tracim/tracim/lib/webdav/sql_resources.py [hunk 1]
```diff
@@ -215,14 +215,24 @@ def createEmptyResource(self, file_name: str):
...
+            # INFO - G.M - 21-03-2018 create new empty file and commit it.
+            new_empty_file = FakeFileStream(
+                file_name=file_name,
+                content_api=self.content_api,
+                workspace=self.workspace,
+                content=content,
+                parent=self.content,
+                path=path,
+            )
+            new_empty_file.close()
```
Comment (buxx) [inline tracim/tracim/lib/webdav/sql_resources.py]: Is it necessary to use FakeFileStream ? If the goal is to create a Content, maybe just use ContentLib ?

File: tracim/tracim/lib/webdav/sql_resources.py [hunk 2]
```diff
@@ -215,14 +215,24 @@ def createEmptyResource(self, file_name: str):
...
+
+        # INFO - G.M - 21-03-2018 - Obtain true davFile from provider
+        resource = self.provider.getResourceInst(path, self.environ)
+        if not resource:
+            raise DAVError(HTTP_FORBIDDEN)
+        content = resource.content
+        return File(path=path, content=content,environ=self.environ)
```
Comment (buxx) [inline tracim/tracim/lib/webdav/sql_resources.py]: missing space here: `content=content,Xenviron=`

File: tracim/tracim/lib/webdav/sql_resources.py [hunk 3]
```diff
@@ -215,14 +215,24 @@ def createEmptyResource(self, file_name: str):
...
+                content=content,
+                parent=self.content,
+                path=path,
+            )
+            new_empty_file.close()
+
+        # INFO - G.M - 21-03-2018 - Obtain true davFile from provider
+        resource = self.provider.getResourceInst(path, self.environ)
+        if not resource:
+            raise DAVError(HTTP_FORBIDDEN)
```
Comment (buxx) [inline tracim/tracim/lib/webdav/sql_resources.py]: Why return `HTTP_FORBIDDEN` instead 404 ?

File: tracim/tracim/lib/webdav/sql_resources.py (hunk 1 above)
Comment (inkhey) [reply tracim/tracim/lib/webdav/sql_resources.py]: I'm don't think so, this MR is actually more a proof of concept, i will refactor this. i'm thinking about removing all or most FakeFileStream class  and add his code directly into File (dav_resource).

Comment (buxx) [reply tracim/tracim/lib/webdav/sql_resources.py]: Ok. So open an issue for that.

**Pull request merged**